### PR TITLE
Update URL schema for filterjitsu search query params

### PR DESCRIPTION
**Pull request by marcusmolchany**
Closes https://github.com/Fullscreen/filterjitsu/issues/9

### CODE REVIEW

**Comment by claudiofullscreen** [comment]
@marcusmolchany the equal sign is a [reserved character for URLs](https://en.wikipedia.org/wiki/Percent-encoding#Types_of_URI_characters) and:

> When a character from the reserved set (a "reserved character") has special meaning (a "reserved purpose") in a certain context, and a URI scheme says that it is necessary to use that character for some other purpose, then the character must be percent-encoded. 

Therefore, the `==` should actually be `%3D%3D`.

In general, every character in your filter should be percent-encoded if necessary.

As an example, imagine that one of your categories includes an `&` sign (e.g. `Rock&Roll`). 
With the current PR, the URL to only show `Rock&Roll` would look like this:

`index.html?filter=category==Rock&Roll`

In this case, your browser would interpret the URL to have two parameters:

- key: `filter`, value: `'category==Rock'`
- key: `Roll`, value: `''`.

The correct way would be `index.html?filter=category%3D%3DRock%26Roll`

**Comment by philipqnguyen** [comment]
This might be useful:

Encode:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent

Decode:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/decodeURIComponent

**Comment by marcusmolchany** [comment]
thanks @philipqnguyen think this is the cleanest solution
```js
var searchParams = decodeURIComponent(window.location.search).replace('?', '').split('&')
```

then we can keep the rest of the code the same

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,10 @@ For more information about changelogs, check
 [Keep a Changelog](http://keepachangelog.com) and
 [Vandamme](http://tech-angels.github.io/vandamme).
 
+## 2.0.0 - 2016.08.22
```
Comment (claudiofullscreen) [inline CHANGELOG.md]: @marcusmolchany you can replace this date with `unreleased` until you decide to release.

**Comment by marcusmolchany** [comment]
woops just saw your comment. i removed it from the changelog entirely. let me know if i should add that back with `unreleased`.

**Comment by marcusmolchany** [comment]
I think this is getting close to complete. Need to make sure all the documentation and tests are in sync and make sense.

Follow up work will be to make the `filter` in `?filter=` a setting for the users to configure.

**Comment by philipqnguyen** [comment]
OK I will incorporate this branch of the filterjitsu library into a test branch on T+ for testing =D

File: dist/jquery.filterjitsu.js [hunk 2]
```diff
@@ -25,7 +25,32 @@
...
+    for (i = 0; i < decodedSearchParams.length; i++) {
+      currParam = decodedSearchParams[i];
+
+      if (currParam.slice(0, 7) === 'filter=') {
+        filterSearchParams = currParam.replace('filter=', '').split(',');
+        break;
+      }
+    }
+
+    return filterSearchParams;
```
Comment (marcusmolchany) [inline dist/jquery.filterjitsu.js]: `filterSearchParams` _could_ return `undefined` if there are no matching search query params. in this case this breaks $.each() on line 63 which expects an array.

File: README.md [hunk 3]
```diff
@@ -27,10 +27,22 @@ initialized. The `DATA_FILTERABLE` selector is required on each item that can be
 that reason it is the best jQuery selector to initialize the plugin with (but this is up to you).
 The filterable items are filtered based on key value pairs provided in the search query parameters of
 the url that match data attributes on the filterable elements. For example, a URL with the search
-query `?type=video&genre=comedy` would filter all elements that did not have the `data-type="video"`
+query `?filter-type=video&genre=comedy` would filter all elements that did not have the `data-type="video"`
```
Comment (marcusmolchany) [inline README.md]: type: genre -> filter-genre

File: README.md [hunk 4]
```diff
@@ -27,10 +27,22 @@ initialized. The `DATA_FILTERABLE` selector is required on each item that can be
...
 or `data-genre="comedy"` data attributes. It is important to note that the url parameters form a
 boolean OR when filtering out filterable elements.
 
+Filterjitsu expects encoded URLs. Encoding can be done using a language library such as those suggested in
+[this stackoverflow post](http://stackoverflow.com/questions/2834034/how-do-i-raw-url-encode-decode-in-javascript-and-ruby-to-get-the-same-values-in/2834053#2834053).
+This is done to percent encode [reserved characters](https://en.wikipedia.org/wiki/Percent-encoding#Types_of_URI_characters) in URLs.
+The unencoded URL format would look as follows:
+```
+?filter-type==music,genre==rock&roll
```
Comment (marcusmolchany) [inline README.md]: typos

File: README.md [hunk 5]
```diff
@@ -27,10 +27,22 @@ initialized. The `DATA_FILTERABLE` selector is required on each item that can be
...
+```
+and after properly encoding it changes to:
+```
+?filter-type=video&genre=rock%26roll
```
Comment (marcusmolchany) [inline README.md]: typos

**Comment by philipqnguyen** [comment]
Sweet. I think this would make it easier for anybody to use.

**Comment by philipqnguyen** [comment]
Let's try it first on T+.

**Comment by marcusmolchany** [comment]
fixed more url typos in the last 2 commits

**Comment by claudiofullscreen** [comment]
Did you try on T+? Working?

**Comment by philipqnguyen** [comment]
I think this branch is ready when you are @marcusmolchany 

@claudiofullscreen the link below is the proof of concept showing the new URL scheme on T+:

https://trailerplus-production-pr-287.herokuapp.com/snapshots/2?filter-platform=YouTube&filter-category=international

**Comment by philipqnguyen** [comment]
I think this is ready! It works on T+ @claudiofullscreen

**Comment by marcusmolchany** [comment]
closing in favor of https://github.com/Fullscreen/filterjitsu/pull/27

**Pull request closed without merging**